### PR TITLE
Erdikose/fix/following link cluster view

### PR DESCRIPTION
**Pull request by erdkse**
This PR...

## Changes

-

## Fixes

-

## How to test it

-

## Screenshots

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Comment by erdkse** [comment]
fixes this https://github.com/kubeshop/monokle/issues/3034

**Comment by erdkse** [comment]
https://user-images.githubusercontent.com/2957624/230291516-561479fe-0d10-448a-8c32-c35da2b728aa.mov

File: src/components/organisms/DashboardPane/MenuItem.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      return clusterConnectionOptions.lastNamespaceLoaded === namespace;
+    },
+    [clusterConnectionOptions]
+  );
+
+  useEffect(() => {
+    if (onActiveMenuItem && activeMenu.key === menuItem.key && menuItemRef && menuItemRef.current) {
+      onActiveMenuItem(menuItemRef.current);
+    }
+  }, [activeMenu]);
```
Comment (topliceanurazvan) [inline src/components/organisms/DashboardPane/MenuItem.tsx:53]: Can we fix the warning here?

File: src/components/organisms/DashboardPane/MenuItem.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  type,
+  menuItem,
+  onActiveMenuItem,
+}: {
+  type: 'parent' | 'child';
+  menuItem: DashboardMenu;
+  onActiveMenuItem?: Function;
+}) => {
+  const clusterResourceMeta = useResourceMetaMap('cluster');
+  const clusterConnectionOptions = useAppSelector(state => state.main.clusterConnectionOptions);
```
Comment (topliceanurazvan) [inline src/components/organisms/DashboardPane/MenuItem.tsx:28]: We can use `useRefSelection` for this one since it's not used for rerendering UI

**Review by topliceanurazvan — CHANGES_REQUESTED** [review]

File: src/components/organisms/DashboardPane/MenuItem.tsx (hunk 1 above)
Comment (erdkse) [reply src/components/organisms/DashboardPane/MenuItem.tsx:53]: Done!

File: src/components/organisms/DashboardPane/MenuItem.tsx (hunk 2 above)
Comment (erdkse) [reply src/components/organisms/DashboardPane/MenuItem.tsx:28]: Done!

**Review by topliceanurazvan — APPROVED** [review]

**Pull request merged**